lib/graphql/field/resolve.rb

```diff
@@ -51,7 +61,11 @@ def initialize(field)
         end
 
         def call(obj, args, ctx)
-          obj.public_send(@field.name)
+          if method_with_arguments?(@field.name, obj, args)
+            obj.public_send(@field.name, args, ctx)
+          else
+            obj.public_send(@field.name)
+          end
```

Comment:
I expected to find a noticeable performance regression from this addition but it was very small (within the margin of error, so maybe nothing) 

```
~/code/graphql $ git co passing-arguments-to-root-fields
Switched to branch 'passing-arguments-to-root-fields'
Your branch is up-to-date with 'f/passing-arguments-to-root-fields'.
~/code/graphql $ be rake bench:query
Warming up --------------------------------------
               query     3.000  i/100ms
Calculating -------------------------------------
               query     38.469  (± 7.8%) i/s -    192.000  in   5.029717s
~/code/graphql $ be rake bench:query
Warming up --------------------------------------
               query     4.000  i/100ms
Calculating -------------------------------------
               query     39.447  (± 2.5%) i/s -    200.000  in   5.076236s
~/code/graphql $ be rake bench:query
Warming up --------------------------------------
               query     3.000  i/100ms
Calculating -------------------------------------
               query     35.082  (±20.0%) i/s -    168.000  in   5.036288s
~/code/graphql $ git co master
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
~/code/graphql $ be rake bench:query
Warming up --------------------------------------
               query     3.000  i/100ms
Calculating -------------------------------------
               query     39.593  (± 5.1%) i/s -    198.000  in   5.008667s
~/code/graphql $ be rake bench:query
Warming up --------------------------------------
               query     4.000  i/100ms
Calculating -------------------------------------
               query     40.610  (± 4.9%) i/s -    204.000  in   5.031973s
~/code/graphql $ be rake bench:query
Warming up --------------------------------------
               query     3.000  i/100ms
Calculating -------------------------------------
               query     40.077  (± 5.0%) i/s -    201.000  in   5.023988s
```